gammapy/datasets/spectrum.py

```diff
@@ -370,9 +367,14 @@ def plot_counts(self, ax=None):
         import matplotlib.pyplot as plt
 
         ax = plt.gca() if ax is None else ax
-
-        self.npred_sig().plot(ax=ax, label="mu_src")
-        self.excess.plot(ax=ax, label="Excess")
+        self.excess.plot(
+            ax=ax, label="Measured excess", yerr=np.sqrt(self.excess.data.flatten())
+        )
+        self.npred_sig().plot(
```

Comment:
I would not put errors on the predicted signal. You could either use points or a continuous line (e.g. an histogram line).